test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+    await proveTx(cardPaymentProcessor.whitelist(account.address));
+  }
+
+  async function clearPayments(payments: TestPayment[], adminAccount: SignerWithAddress) {
+    const authorizationIds: string[] = [];
+    payments.forEach(payment => {
+      authorizationIds.push(createBytesString(payment.authorizationId, BYTES16_LENGTH));
+      payment.status = PaymentStatus.Cleared;
+    });
+    await proveTx(cardPaymentProcessor.connect(adminAccount).clearPayments(authorizationIds));
+  }
+
+  function defineBalancesPerAccount(payments: TestPayment[], targetPaymentStatus: PaymentStatus) {
+    const balancesPerAccount: Map<string, number> = new Map<string, number>();
+
+    payments.forEach(payment => {
+      const address: string = payment.account.address;
+      let newBalance: number = balancesPerAccount.get(address) || 0;
+      if (payment.status == targetPaymentStatus) {
+        newBalance += payment.amount;
+      }
+      balancesPerAccount.set(address, newBalance);
+    });
+
+    return balancesPerAccount;
+  }
+
+  function defineExpectedCardPaymentProcessorState(payments: TestPayment[]): CardPaymentProcessorState {
+    const tokenBalance: number = payments.map(function (payment: TestPayment): number {
+      return payment.status == PaymentStatus.Uncleared || payment.status == PaymentStatus.Cleared ? payment.amount : 0
+    }).reduce((sum: number, current: number) => sum + current);
+    const totalClearedBalance: number = payments.map(function (payment: TestPayment): number {
+      return payment.status == PaymentStatus.Cleared ? payment.amount : 0
+    }).reduce((sum: number, current: number) => sum + current);
+    const totalUnclearedBalance: number = payments.map(function (payment: TestPayment): number {
+      return payment.status == PaymentStatus.Uncleared ? payment.amount : 0
+    }).reduce((sum: number, current: number) => sum + current);
+    const clearedBalancesPerAccount: Map<string, number> = defineBalancesPerAccount(
+      payments,
+      PaymentStatus.Cleared
+    );
+    const unclearedBalancesPerAccount: Map<string, number> = defineBalancesPerAccount(
+      payments,
+      PaymentStatus.Uncleared
+    );
+
+    return {
+      tokenBalance,
+      totalUnclearedBalance,
+      totalClearedBalance,
+      clearedBalancesPerAccount,
+      unclearedBalancesPerAccount
+    };
+  }
+
+  async function checkPaymentStructuresOnBlockchain(payments: TestPayment[]) {
+    for (let i = 0; i < payments.length; ++i) {
+      const expectedPayment: TestPayment = payments[i];
+      const actualPayment = await cardPaymentProcessor.paymentFor(
+        createBytesString(expectedPayment.authorizationId, BYTES16_LENGTH)
+      );
+      checkEquality(actualPayment, expectedPayment, i);
+      if (!!expectedPayment.parentTxHash) {
+        expect(await cardPaymentProcessor.isPaymentReversed(expectedPayment.parentTxHash)).to.equal(
+          expectedPayment.status == PaymentStatus.Reversed
+        );
```

Comment:
Codacy has a fix for the issue: [Replace `);` with `expect(`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
        expect(
```